services/app-api/src/postgres/contractAndRates/insertContract.ts

```diff
@@ -1,20 +1,56 @@
 import { PrismaClient } from '@prisma/client'
 import { v4 as uuidv4 } from 'uuid'
 import { Contract } from './contractType'
+import { ContractFormData } from '../prismaTypes'
+
+async function incrementAndGetStateNumber(
+    client: PrismaClient,
+    stateCode: string
+): Promise<number | Error> {
+    try {
+        const stateNumberResult = await client.state.update({
+            data: {
+                latestStateSubmissionNumber: {
+                    increment: 1,
+                },
+            },
+            where: {
+                stateCode: stateCode,
+            },
+        })
+
+        return stateNumberResult.latestStateSubmissionNumber
+    } catch (e) {
+        return e
+    }
+}
 
 // creates a new contract, with a new revision
 async function insertDraftContract(
     client: PrismaClient,
-    formData: string
+    formData: ContractFormData
```

Comment:
instead of accepting a whole ContractFormData type here, let's only accept the fields that are required. See how it's working in insertHPP